### ISSUE TITLE
Refactor to make WebExtensionSupport.initalize() simpler

### ISSUE THREAD

**Issue by psymoon** [mozilla-mobile/android-components#5647]
We have many lambdas in the parameter (1-2 more to come), can we simplify by introducing a `WebExtensionUseCase`?:
```
WebExtensionSupport.initialize(
    components.engine,
    components.store,
    onNewTabOverride = {
        _, engineSession, url ->
            val session = Session(url)
            components.sessionManager.add(session, true, engineSession)
            session.id
    },
    onCloseTabOverride = {
        _, sessionId -> components.tabsUseCases.removeTab(sessionId)
    },
    onSelectTabOverride = {
        _, sessionId ->
            val selected = components.sessionManager.findSessionById(sessionId)
            selected?.let { components.tabsUseCases.selectTab(it) }
    },
    onUpdatePermissionRequest = components.addonUpdater::onUpdatePermissionRequest
)

**Comment by csadilek** [comment]
Note that `WebExtensionSupport` is a temporary component that is only meant to exist until our browser-state refactoring is complete. Once that is done, the engine will have access to the store and this component is no longer needed. It's definitely possible to make this simpler and provide overrideable use cases (for deciding whether or not browser-session should be used), but I think we should only do this if the effort is small. I'd like to get rid of this class pretty soon :).

**Comment by csadilek** [comment]
We came up with a new design for the tab vs popup problem for extension actions and will remove the lambda. Since we want to remove this class when the browser-state refactoring is complete there's no value in refactoring the lambdas.
